### PR TITLE
MP-1959 add license

### PR DESCRIPTION
**Pull request by larry0x**
Add GPL-v3-or-later license.

I've also taken this opportunity to update package metadata (use the workspace feature) and update Cargo.lock dependencies.

### CODE REVIEW

**Review by piobab — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,16 @@
 [workspace]
 members = ["contracts/*", "packages/*"]
 
+[workspace.package]
+version       = "0.1.0"
+authors       = ["Larry Engineer <larry@delphidigital.io>"]
+edition       = "2021"
+rust-version  = "1.65"
```
Comment (piobab) [inline Cargo.toml:8]: 1.66.1?

Comment (piobab) [reply Cargo.toml:8]: https://blog.rust-lang.org/2023/01/10/Rust-1.66.1.html
https://blog.rust-lang.org/2023/01/10/cve-2022-46176.html

Comment (larry0x) [reply Cargo.toml:8]: 1.65 here means the minimum supported version. The workspace feature is only available since 1.65, so 1.65 is the minimum.

Comment (piobab) [reply Cargo.toml:8]: btw accroding to rust version, I wonder if we should specify exact rust version instead of working on newest one, wdyt?

Comment (larry0x) [reply Cargo.toml:8]: For consistency I perhaps should just use whichever rust version the latest rust-optimizer uses.

**Pull request merged**